core/modules/utils/dom/dom.js

```diff
@@ -311,6 +311,48 @@ exports.getLocationPath = function() {
 	return window.location.toString().split("#")[0];
 };
 
+exports.copyEventProperties = function(event) {
+	var seen = new Set();
+  
+	function isDOMElement(value) {
+	  return value instanceof Node || value instanceof Window;
+	}
+  
+	function safeCopy(obj) {
+		//skip ciruclar references
+		if(seen.has(obj)) {
+			return "[Circular reference]";
+		}
+		//skip functions
+		if(typeof obj !== "object" || obj === null) {
+			return obj;
+		}
+		//skip DOM elements
+		if(isDOMElement(obj)) {
+			return "[DOM Element]";
+		}
+		//copy each element of the array
+		if(Array.isArray(obj)) {
+			return obj.map(safeCopy);
+		}
+
+		seen.add(obj);
+		var copy = {}, key;
+		for(key in obj) {
+			try{
```

Comment:
It's a shame we need all these nested try/catch constructions. I assume that the performance impact is not significant?

Reply:
In all the testing I have done, the time for this function to return a result from an event object has ranged between 0.004 and 0.1ms, so I do not think the performance impact is significant at all.

Currently we return strings indicating what the skipped element was, such as "[DOM Element]". If we just want to skip circular references and DOM references without returning a hint as to what they are, we can combine those checks.